script.js

```diff
@@ -0,0 +1,196 @@
+    'coth(': '1/Math.tanh(',
+}
+var result = '';
+var output;
+var ans = document.getElementById("ans");
+
+function factorial(data) {
+    let fact = 1;
+    for (let i = 1; i <= data; i++) {
+        fact = fact * i;
+    }
+    return fact;
+}
+function putdata(data) {
+    input.value += data;
+}
+function storeinMemory() {
+    console.log(result);
+}
+
+function clearText() {
+    input.value = "";
+    result = '';
+    ans.innerHTML = "Output : 0";
+}
+function deleteChar() {
+    let str = input.value;
+    str = str.slice(0, -1);
+    input.value = str;
+    result = result.slice(0, -1);
+    if (input.value.length == 0) {
+        result = ''
+    }
+}
+
+function showAns() {
+    let count = 0, pattern1 = /[\(]/g, pattern2 = /[\)]/g;
+    try {
+        let op, inp = input.value;
+        for (let x in values) {
+            if (inp.includes(x)) {
+                inp = inp.replaceAll(x, values[x]);
+            }
+        }
+        if (inp.includes('E')) {
+            inp = inp.replaceAll('E', 2.718);
+        }
+        op = inp;
+        count = op.replace(pattern1, "").length;
+        if (count != op.replace(pattern2, "").length) {
+            throw "*Put valid paranthesis.";
+        }
+        output = eval(op);
+        ans.innerHTML = "Output : " + output;
+        document.getElementById("error").innerHTML = "";
+        input.value = ''; 
+    }
+    catch (e) {
+        document.getElementById("error").innerHTML = "*Enter valid expression.<br>" + e;
+    }
+
+}
+
+// Memory function
+function storeinMemory(value) {
+    let data;
+    if (value == "MS") {
+        localStorage.setItem('calcdata', input.value);
+    }
+    if (value == "MC") {
+        localStorage.clear();
+    }
+    if (value == "MR") {
+        data = localStorage.getItem('calcdata');
+        ans.innerHTML = "Memory Recall : " + data;
+    }
+    if (value == "M+") {
+        let exp;
+        data = localStorage.getItem('calcdata');
+        exp = input.value + '+' + data;
+        input.value = eval(exp);
+        ans.innerHTML = "Memory Value Addition : " + eval(exp);
+    }
+    if (value == "M-") {
+        let exp;
+        data = localStorage.getItem('calcdata');
+        exp = input.value + '-' + data;
+        input.value = eval(exp);
+        ans.innerHTML = "Memory Value Subtraction : " + eval(exp);
+    }
+}
+
+// All toggle menus
+function second_menu() {
+    let first_class = document.getElementsByClassName("first");
+    let second_class = document.getElementsByClassName("second");
+    let style = getComputedStyle(document.querySelector(".first")).display;
+    if (style == "inline-block") {
+        for (let i = 0; i < 6; i++) {
+            first_class[i].style.display = "none";
+            second_class[i].style.display = "inline-block";
+        }
+    }
+    else {
+        for (let i = 0; i < 6; i++) {
+            first_class[i].style.display = "inline-block";
+            second_class[i].style.display = "none";
```

Comment:
wrap this code inside one loop and check the condition inside the loop so it will result in the use of a single loop.
it only for reduce LOC